pkg/cri/server/image_pull.go

```diff
@@ -154,6 +159,32 @@ func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest)
 		tracing.Attribute("image.ref", ref),
 		tracing.Attribute("snapshotter.name", snapshotter),
 	)
+
+	// Pull image using transfer service
+	if c.config.PluginConfig.ImagePullWithTransferService {
+
+		var sopts []transferimage.StoreOpt
+
+		sopts = append(sopts, transferimage.WithPlatforms(platforms.DefaultSpec()))
+		sopts = append(sopts, transferimage.WithUnpack(platforms.DefaultSpec(), snapshotter))
+		sopts = append(sopts, transferimage.WithImageLabels(map[string]string{imageLabelKey: imageLabelValue}))
+
+		ch, _ := newcriCredentials(ctx, ref, r.GetAuth())
+		reg := registry.NewOCIRegistry(ref, nil, ch)
+		is := transferimage.NewStore(ref, sopts...)
+
+		// pf, done := ProgressHandler(ctx, os.Stdout)
+		// defer done()
+		//Todo handle procress
+		err = c.client.TransferService().Transfer(ctx, reg, is)
+		if err != nil {
+			return nil, fmt.Errorf("failed to pull and unpack image %q: %w", ref, err)
+		}
+
+		return &runtime.PullImageResponse{ImageRef: ref}, nil
```

Comment:
need to update the image store.

Reply:
Done.
Just curious CRI currently maintain an in-memory image cache(there could be situations that the CRI image store and containerd image store are out of sync, like other clients e.g. ctr removed an image). Can we get rid of it by directly using contained image store?